packages/core/src/__tests__/ceramic-query-response.test.ts

```diff
@@ -0,0 +1,123 @@
+import { jest, expect, describe, test, beforeEach, afterEach } from '@jest/globals'
+import { type CeramicApi, type IpfsApi, TestUtils } from '@ceramicnetwork/common'
+import { TileDocument } from '@ceramicnetwork/stream-tile'
+import { StreamID } from '@ceramicnetwork/streamid'
+import { createIPFS, swarmConnect } from '@ceramicnetwork/ipfs-daemon'
+import { createCeramic } from './create-ceramic.js'
+import type { SignedMessage } from '@libp2p/interface-pubsub'
+import { deserialize, MsgType, serialize } from '../pubsub/pubsub-message.js'
+import { MAX_RESPONSE_INTERVAL } from '../pubsub/message-bus.js'
+
+const LONG_SYNC_TIME = 60 * 1000
+
+describe('Response to pubsub queries handling', () => {
+  jest.setTimeout(30 * 1000)
+
+  let ceramicIpfs: IpfsApi
+  let ceramic: CeramicApi
+  let otherIpfs: IpfsApi
+  beforeEach(async () => {
+    ceramicIpfs = await createIPFS()
+    ceramic = await createCeramic(ceramicIpfs)
+    otherIpfs = await createIPFS()
+
+    await swarmConnect(ceramicIpfs, otherIpfs)
+  })
+
+  afterEach(async () => {
+    await otherIpfs.pubsub.unsubscribe(ceramic.topic)
+    await ceramic.close()
+    await ceramicIpfs.stop()
+  })
+
+  test('sync returns after only response', async () => {
+    const genesisCommit = await TileDocument.makeGenesis(ceramic, { foo: 'bar' }, null)
+    const genesisCID = await ceramic.dispatcher.storeCommit(genesisCommit)
+    const streamID = new StreamID(0, genesisCID)
+
+    const timeBeforeResponding = 100
+    await otherIpfs.pubsub.subscribe(ceramic.pubsubTopic, async (rawMessage: SignedMessage) => {
+      const message = deserialize(rawMessage)
+      if (message.typ == MsgType.QUERY && message.stream.equals(streamID)) {
+        await TestUtils.delay(timeBeforeResponding)
```

Comment:
I'd love to be able to write this test file without using a second real IPFS node but instead by injecting pubsub messages into the IncomingChannel at the time rate that I want.  I think it would make the tests lighter-weight and more reliable, instead of depending on a real pubsub connection.  But I couldn't figure out how to do that.  Please LMK if you think that is relatively straightforward to do.

Reply:
You could attach a jest spy to ceramicIpfs instance's `subscribe`, that calls subscription handler at any rate you want after it gets a query. The latter part again could be achieved by a jest spy.

Reply:
I tried to get this to work but failed :(
You can see my attempt here if you like: https://github.com/ceramicnetwork/js-ceramic/blob/tipResponseTestRewriteWithJestMocks/packages/core/src/__tests__/ceramic-query-response.test.ts#L114.  If you have any suggestions on how to fix it I'm happy to try again, otherwise I'll probably just merge this PR as-is tomorrow.

Reply:
Ooh good call, thanks!  Got the tests all updated and switched over to injecting pubsub responses directly instead of actually using a real ipfs pubsub connection.  PTAL